### DISCUSSION TITLE
Remove the glow-up effect that makes a note brighter just before hitting it

### DISCUSSION THREAD

**Discussion by isakvik** [ppy/osu#25745] [Ideas]
### Type

Game behaviour

### Bug description

About 300ms before the timing of a hitcircle, it approaches full brightness from a level that's distinctly lower. It should go without saying that while this is well and good on any usually configured monitor, lowering gamma makes it a visual cue that's effortless to use for bypassing reading difficulty on high density maps.

Cannot spot it in the video? Turn your gamma down and play an AR0 map, it'll be clear as day.

### Screenshots or videos

https://github.com/ppy/osu/assets/11044829/4bff49b9-6b25-4348-8076-e1730eb755f4

### Version

2023.1130.0

### Logs

none

**Comment by peppy** [comment]
This has been around forever and is unlikely to change.

**Reply by isakvik** [reply]
This is a game integrity issue that lets players ignore a beatmap's AR by fiddling with settings and you're seriously considering keeping it around? Why not just let difficulty adjust give pp to begin with if we're going down that road? 

Edit: by forever you mean it's also in stable, but since the circles are even darker before fading in on lazer it's also subject to abuse to a higher degree. I'd like to see the whole feature go, but if you're gonna be adamant about keeping it, tone down the effect - harshly.

Edit 2: Additional complication - looks like this hasn't been implemented the same way between stable and lazer; one multiplier is applied to the whole of the hitobject in lazer, while stable has one applied to the hitcircle fill, and a separate to the hitcircleoverlay? Reference screenshot - outlines are pretty dark on lazer:

![image](https://github.com/ppy/osu/assets/11044829/e547ecce-cdf1-4c08-bb82-45353dfe54f9)

**Reply by Walavouchey** [reply]
<details><summary>comparison of default skin, first on stable then lazer</summary>

![screenshot744](https://github.com/ppy/osu/assets/36758269/bc7f6b12-e00d-41c8-a93e-abcd2a81b697)

![osu_2023-12-13_10-14-23](https://github.com/ppy/osu/assets/36758269/b4999626-81b4-4cff-b6d6-bc2d411e5068)

</details>

there's no difference in the outline and hit circle body (the colour multiplier value is exactly the same - (195, 195, 195) or ~76% brightness), but the numbers are clearly brighter in stable (in case it's not clear, try opening the images in separate tabs and flipping between them)

**Reply by isakvik** [reply]
Yeah I can't reproduce a difference between stable and lazer with the default skin, so the difference then has to be some sort of skin versioning issue or similar, as far as I can tell (the hitcircleoverlay I use is lifted directly from stable's default).

[_GN (-GN_ eren-desu_ Yasen etc.).zip (lazer skin export)](https://github.com/ppy/osu/files/13659442/_GN.-GN_.eren-desu_.Yasen.etc.zip)

[reference screenshot of lazer default skin](https://github.com/ppy/osu/assets/11044829/ac1b7c7f-49c1-4d28-985b-fb6cc6acf12e)

**Reply by bdach** [reply]
The hitcircle number discrepancy is well known and thus far has been accepted by us as an intentional change due to stable's behaviour being idiosyncratic at best. See description in https://github.com/ppy/osu/pull/24968.

I haven't looked in detail into the skin provided - if anything here is actionable, it's that first and foremost.

**Reply by peppy** [reply]
I was waiting for this *specific case* to be brought up, and even cited it IRL. It only affects skins which (ab)use combo number sprites for other purposes.

I even said that "these players are either going to love or hate this".

We could match stable on this, but it's not really making anything more or less "exploitable". I'll stress again, the dim matches stable in every way and the only difference is that it's applied to the combo number sprite in lazer but not stable. So at the end of the day, depending on the structure of your skin, you can achieve the precise same effect.

**Comment by bdach** [comment]
I see the visual effect as an effective visual cue as to which objects are approaching temporally. It is a visual aid.

So which outcome is better here?

- Removing the visual aid, which will hinder new players foremost (but realistically affects the whole playerbase universally)
- Allowing the visual aid, which can lead to fairness disputes along the upper competitive echelon of players who mess with their monitor settings (affects a small percentage of players)

I cannot shake the feeling that going with the latter option is pouring the baby out with the bathwater, so to speak. osu! is not the only game in which people may want to abuse gamma. If you go further, most esports have had controversies and accusations related to non-standard setups and custom hardware that in opinion of some do or don't provide a competitive advantage ([example 1](https://kotaku.com/smash-god-drops-out-of-tournament-because-his-controlle-1794769487), [example 2](https://www.esports.net/news/counter-strike/astralis-cheating-smoke-settings-drama/), off the top of my head). I see trying to curb this as unrealistic.

**Reply by Bubbleman532** [reply]
In both examples though it's a fault of the game that means it can be exploited. See TaikoMania skins for another example of a simple exploit that makes top level play trivial, which I'm aware has been patched out.

I wager no beginners even fully notice the fade-in since having to focus on timing and cursor movement are such new concepts that it's all that really occupies the active thought process, but that's just my viewpoint and I have no data to back that up. I also know that there was a patch applied to stable to remove or at the very least mitigate this issue which I'm sure the reading players will be better versed on.

**Reply by yokespai** [reply]
As a side note to Bubbleman's comment, the tutorial supplied with the game makes any new player focus on the approach circle as the go-to visual aid, meaning that new players will be focusing on using said approach circle as the visual aid and most likely miss the glow-up effect.

Your argument and feeling lose credibility when you realize the game already has a main visual aid (again, from the tutorial), which diminishes what you think will hinder new players, in addition to that some players may never notice this subtle aid in the first place. For example, as a player with near 1700 hours on stable, I never noticed a glow-up on lazer when I tried it, nor did I notice the glow-up on stable before it was patched out in 2022 (when I started in 2015). I found out about it through someone else mentioning it.

**Reply by isakvik** [reply]
It's a visual aid that quite effectively bypasses a whole tier of difficulty setting values if abused, applied globally. That's a pretty large cost to pay for a bandaid aimed at new players' subconscious who can't deal with visual clutter effectively, at least until they've learned to follow the main visual cue used to gauge timings (the approach circle). Like, gamma abuse is such an onerous aspect of this game that [peppy has threatened to essentially remove a reading-based arena of competition because of it](https://x.com/ppy/status/1497054806862856200) (granted - lazer offers several new ones, given that it keeps the integrity of its game mechanics!), but leaving this a possibility here is just an annoyance for both players who intend to use it as well as players who respect the difficulty of the game as they enjoy it.

It's also abusable in stable, by the way, but it takes some real effort to make it viable. Most, though not all, of my shock and horror comes from the excessive dimming seen in the OP (that happens for custom skins only?), and as discussed [above](https://github.com/ppy/osu/discussions/25745#discussioncomment-7839229).

**Reply by Nyanaro** [reply]
Calling it a visual aid is quite generous (an aesthetic effect would be more appropriate). It is completely unnoticeable to the average player, and is unreactable for the average player (300ms is the same as approach rate 11). While the effect is also in stable, the effect is far more pronounced in Lazer on affected skins (which new players wouldn't be using), and only leaves room for exploitation.
